src/test/app/Path_test.cpp

```diff
@@ -171,6 +171,131 @@ rpf(jtx::Account const& src, jtx::Account const& dst, std::uint32_t num_src)
     return jv;
 }
 
+// test helper
+bool checkArraySize(Json::Value const& val, unsigned int size)
+{
+    return val.isArray() &&
+           val.size() == size;
+}
+
+using verify_paths =
+    std::vector<std::vector<std::pair<ripple::Issue, jtx::Account>>>;
+
+// compares expected paths (pairs of Issue/Account) with
+// json output from path_find
+bool
+check_paths(
```

Comment:
check_paths compares json (from the API) to expectations which are written in terms or Amount/Issue. I think you're right, though, that I ought to look more closely at the existing find_paths/same mechanism that is already there to see if I can use it to make the comparisons instead. I was initially disinclined to go messing with existing comparison code, but I think it's worth a second look.